sys/net/include/etx_beaconing.h

```diff
@@ -94,25 +97,49 @@ typedef struct __attribute__((packed)) etx_probe_t {
 } etx_probe_t;
 
 typedef struct etx_neighbor_t {
-    ipv6_addr_t addr;           //The address of this node
-    uint8_t     tx_cur_round;   //The indicator for receiving a packet from this candidate this round
-    uint8_t     packets_tx[ETX_WINDOW]; //The packets this node has transmitted TO ME
-    uint8_t     packets_rx;     //The packets this node has received FROM ME
-    double      cur_etx;        //The currently calculated ETX-value
-    uint8_t     used;           //The indicator if this node is active or not
+    ipv6_addr_t addr;           /** The address of this node */
+    uint8_t     tx_cur_round;   /** The indicator for receiving a packet from this candidate this round */
+    uint8_t     packets_tx[ETX_WINDOW]; /** The packets this node has transmitted TO ME */
+    uint8_t     packets_rx;     /** The packets this node has received FROM ME */
+    double      cur_etx;        /** The currently calculated ETX-value */
+    uint8_t     used;           /** The indicator if this node is active or not */
 } etx_neighbor_t;
 
 //prototypes
+
+ /**
+* This function is used to initialise and accepts one parameter . 
+* @param address a ipv6_addr_t pointer.
+* @see etx_get_metric()
+* @see etx_update()
+* @return It does not return any value.
+*/
 void etx_init_beaconing(ipv6_addr_t *address);
+
+ /**
+* This function is used to get the calculated ETX-value.
+* @param address a ipv6_addr_t pointer.
+* @see etx_init_beaconing()
+* @see etx_update()
+* @return It returns the calculated ETX-value.
+*/
 double etx_get_metric(ipv6_addr_t *address);
+
+ /**
+* This function is used to Update the current ETX value of a candidate.
+* @param neighbour an etx_neighbour_t pointer.
```

Comment:
write `@param[in]` for input parameters, `@param[out]` for output parameters, or `@param[inout]` for both.

This is especially important for pointer arguments as well as reference arguments (C++ only)